### PR TITLE
added model for detecting sentiment of claim towards its topic

### PR DESCRIPTION
**Pull request by nelliemirz**
# BERT for detecting sentiment towards the topic of the claim 
We adapted the code from [BERT-for-RCC-ABSA](https://github.com/howardhsu/BERT-for-RRC-ABSA) for our task.

**Task description**:

Given a claim ("Violent video games increase the violent tendencies among youth") and its target ("Violent video games"), detect the sentiment of the claim towards the topic (negative).

The code is tested on Ubuntu 18.04 with Python 3.6.8(Anaconda), PyTorch 1.0.1 and [pytorch-pretrained-bert](https://github.com/huggingface/pytorch-pretrained-BERT) 0.4. 

## Fine-tuning Setup

1) The pre-trained model is saved in the folder ```pt_model```.

2) The model weights that were used for fine-tuning can be found [here](https://drive.google.com/file/d/1io-_zVW3sE6AbKgHZND4Snwh-wi32L4K/view?usp=sharing).

TODO: fine-tune on the model found [here](https://drive.google.com/file/d/1TYk7zOoVEO8Isa6iP0cNtdDFAUlpnTyz/view?usp=sharing). 
And other BERT weights (e.g., bert-base, BERT-DK ([laptop](https://drive.google.com/file/d/1TRjvi9g3ex7FrS2ospUQvF58b11z0sw7/view?usp=sharing), [restaurant](https://drive.google.com/file/d/1nS8FsHB2d-s-ue5sDaWMnc5s2U1FlcMT/view?usp=sharing)) in our paper). 
Make sure to add an entry into ```src/modelconfig.py```.

### Create a dataset

4) Run the script ```create_dataset_sentiment.py``` to create the dataset for the task.

### Training 
5) All the logs for training and testing are stored in ```run```.
6) Fire a fine-tuning from a BERT weight, e.g.
```
cd script
bash run_absa.sh asc laptop_pt claimsentim pt_asc 10 0
```
Here ```asc``` is the task to run, ```laptop_pt``` is the post-trained weights for laptop, ```claimsentim``` is the dataset we want to fine-tune on, ```pt_asc``` is the fine-tuned folder in ```run/```, ```10``` means run 10 times and ```0``` means use gpu-0.

### Evaluation

**ASC**: run the script (returns accuracy and f_macro scores)

```
eval_claimsentim.py /
--task asc --bert pt --domain claimsentim --runs 10
```

### CODE REVIEW

**Review by Bhuvanesh-Verma — APPROVED** [review]

**Comment by Bhuvanesh-Verma** [comment]
Before we merge, maybe we'll try to integrate it into allennlp config. If we get good scores then we use allennlp configs else we try to create python scripts to run it instead of bash scripts. But for now let's keep it as it is.

**Comment by nelliemirz** [comment]
Integrated into allennlp config.
